member/src/main/scala/com/improving/app/member/api/MemberServiceImpl.scala

```diff
@@ -0,0 +1,128 @@
+package com.improving.app.member.api
+import akka.actor.typed.ActorSystem
+import akka.cluster.sharding.typed.scaladsl.{ClusterSharding, Entity}
+import akka.cluster.typed.{Cluster, Join}
+import akka.pattern.StatusReply
+import akka.serialization.jackson.JacksonObjectMapperProvider
+import akka.util.Timeout
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.improving.app.member.domain.Member.{MemberCommand, MemberEntityKey}
+import com.improving.app.member.domain._
+import com.improving.app.member.utils.serialize.AvroJacksonObjectMapperFactory.applyObjectMapperMixins
+import wvlet.airframe.ulid.ULID
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+
+class MemberServiceImpl(implicit val system: ActorSystem[_]) extends MemberService {
+
+  implicit val ec: ExecutionContext = system.executionContext
+  implicit val timeout: Timeout = Timeout(5 seconds)
+
+  //Create a new member
+  val sharding: ClusterSharding = ClusterSharding(system)
+  ClusterSharding(system).init(
+    Entity(MemberEntityKey)(entityContext => Member(entityContext.entityTypeKey.name, entityContext.entityId))
+  )
+
+  Cluster(system).manager ! Join(Cluster(system).selfMember.address)
+
+  implicit val objectMapper: ObjectMapper = JacksonObjectMapperProvider(system).getOrCreate("jackson-cbor", None)
+  applyObjectMapperMixins(objectMapper)
+
+  //Do not use for RegisterMember
+  private def extractEntityId(request: MemberRequest): String = {
```

Comment:
I had also thought of using a trait. The issue is RegisterMember request should not have a memberId as it should be generated by the service. I could add it in there and check if its supplied to the service registerMember endpoint - in that case I can reject the RegisterMember command.

Reply:
For now keeping it out of MemberRequest due to the limitation with RegisterMember not having memberId

Reply:
So RegisterMember is different, all the other commands can share the MemberRequest trait.  RegsiterMember is not a MemberRequest because you need a member for that and RegisterMember is a service request to create a member.

Reply:
I have made all commands except RegisterMember inherit HasMemberId trait - which I use to extractMemberId

Reply:
Great.